collector/collectCommunities.py

```diff
@@ -93,20 +98,40 @@ def summarizedJson(ffDir, path):
         				'User-Agent': 'Mozilla/5.0 (Macintosh; Intel Mac OS X 10_9_3) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/35.0.1916.47 Safari/537.36'
     				}
 			)
-			ffApi = json.loads(urlopen(request, None, 10).readall().decode('utf-8'))
+			ffApi = json.loads(urlopen(request, None, 10).read().decode('utf-8'))	#OLr AttributeError: 'HTTPResponse' object has no attribute 'readall'
 		except UnicodeError as e:
 			try:
-				ffApi = json.loads(urlopen(ffDir[community]).readall().decode('iso8859_2'))
-				log(0, "Unicode Error: " + ffDir[community] + ": " + str(e) + ", try iso8859_2 instead")
-				pass
+				log(0, "Unicode Error: " + ffDir[community] + ": " + str(e) + ", try iso8859_2 instead")	#OLr erst loggen, dann probieren
+				ffApi = json.loads(urlopen(ffDir[community]).read().decode('iso8859_2'))		#OLr AttributeError: 'HTTPResponse' object has no attribute 'readall'
+				#pass
 			except BaseException as e:
+				error = e
 				log(0, "Error reading community api file " + ffDir[community] + ": " + str(e))
-				continue
+				#continue
 		except BaseException as e:
+			error = e
 			log(0, "Error reading community api file " + ffDir[community] + ": " + str(e))
-			continue
+			#continue
+
+		if (error is None):
+			#OLr wie bisher mtime setzten
+			ffApi['mtime'] = time
+		else:
+			#OLr ffApi von summary lesen oder neu erstellen
+			try:	#kein has_key verfügbar
+				ffApi = summary[community]
+			except:
+				ffApi = dict()	#init new instance
+				ffApi['mtime'] = time
+				#nachfolgende Felder werden für tableHtml benötigt
+				ffApi['location'] = dict()
+				ffApi['location']['city'] = community
+				ffApi['name'] = community
+				ffApi['state'] = dict()
+			#OLr im Fehlerfall etime und error setzten (statt mtime)
+			ffApi['etime'] = time
+			ffApi['error'] = str(type(error)) + " " + str(error)	#OLr exception type und nachricht
```

Comment:
falls es dann wieder geht muss man dann den error auch wieder löschen, würde sonst für Verwirrung sorgen

Reply:
Wenn kein Fehler auftritt, überschreibt die neu gelesene API die Alte In der summary. Deshalb keine löschen des 'error' Feldes notwendig.